src/ingestion/sources/bigquery.py

```diff
@@ -0,0 +1,200 @@
+        connection_string_prefix = self._connection_string_prefix()
+        connection_string_base = '{project}/{dataset}'.format(
+            project=configuration.get('project'),
+            dataset=configuration.get('dataset'),
+        )
+        connection_string_params = '?credentials_base64={credentials_base64}'.format(
+            credentials_base64=configuration.get('credentials_base64')
+        )
+
+        connection_string = '{connection_string_prefix}://{connection_string_base}{connection_string_params}'.format(
+            connection_string_prefix=connection_string_prefix,
+            connection_string_base=connection_string_base,
+            connection_string_params=connection_string_params
+        )
+
+        return make_url(connection_string)
+
+
+    def database(self):
+        return json.loads(self.configuration).get("project")
+
+    def schema(self):
+        return json.loads(self.configuration).get('dataset')
+
+    @property
+    def type(self):
+        return "bigquery"
+
+class BigQuerySourceDialect(SQLAlchemySourceDialect):
+    @classmethod
+    def _numeric_std(cls):
+        return "STDDEV(CAST({} as FLOAT64))"
+
+    @classmethod
+    def _std_length(cls):
+        return "STDDEV(CAST(LENGTH({}) as FLOAT64))"
+    
+    @classmethod
+    def _text_int_rate(cls):
+        return "SUM(ARRAY_LENGTH(REGEXP_EXTRACT_ALL(CAST({} AS STRING), r'^([-+]?[0-9]+)$'))) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _text_number_rate(cls):
+        return "SUM(ARRAY_LENGTH(REGEXP_EXTRACT_ALL(CAST({} AS STRING), r'^[\+\-]?\d*\.?\d+(?:[Ee][\+\-]?\d+)?$'))) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _text_uuid_rate(cls):
+        return "SUM(ARRAY_LENGTH(REGEXP_EXTRACT_ALL(CAST({} AS STRING2), r'^([0-9a-fA-F]{{8}}-[0-9a-fA-F]{{4}}-[0-9a-fA-F]{{4}}-[0-9a-fA-F]{{4}}-[0-9a-fA-F]{{12}})$'))) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _text_all_spaces_rate(cls):
+        return "SUM(ARRAY_LENGTH(REGEXP_EXTRACT_ALL(CAST({} AS STRING), r'^\s*$'))) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _text_null_keyword_rate(cls):
+        return "SUM(CASE WHEN UPPER(CAST({} as STRING)) IS NULL THEN 1 ELSE 0 END) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _zero_rate(cls): # TODO: ?
+        return "SUM(CASE WHEN UPPER(CAST({} as STRING)) IS NULL THEN 1 ELSE 0 END) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _negative_rate(cls):
+        return "SUM(CASE WHEN {} < 0 THEN 1 ELSE 0 END) / CAST(COUNT(*) AS NUMERIC)"
+
+    @classmethod
+    def _completeness(cls):
+        return "COUNT({}) / CAST(COUNT(*) AS NUMERIC)"
+    
+    @classmethod
+    def _freshness(cls):
+        return "TIMESTAMP_DIFF(MAX({}), CURRENT_TIMESTAMP, MINUTE)"
+
+    @classmethod
+    def schema_tables_query(cls, database_name, schema_name):
+        raise NotImplementedError
+
+    @classmethod
+    def table_metrics_query(cls, monitor, discovery_data, minutes_ago):
```

Comment:
Why do we need to override this?

Reply:
because we need to use the `BigQueryMetricsQueryBuilder` class for the builder